### PR TITLE
Add description change logging, refs #12130

### PR DESCRIPTION
**Pull request by mcantelon**
* Adds optional logging of creation or modification of descriptions
* Modifies description updates page to show this logging data when logging is
  enabled
* Adds option, for autheticated users, to view a particular description's
  creation/modification history
* Allows administrations to view a history of any user's creation/modification
  of descriptions (and allows users to view this data on their profile)

### CODE REVIEW

File: apps/qubit/modules/informationobject/actions/modificationsAction.class.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  private function abortInvalidRequests()
+  {
+    // Check that this isn't the root
+    if (!isset($this->resource->parent))
+    {
+      $this->forward404();
+    }
+
+    // Check user authorization
+    if (!QubitAcl::check($this->resource, 'read') || !sfConfig::get('app_audit_log_enabled'))
```
Comment (sbreker) [inline apps/qubit/modules/informationobject/actions/modificationsAction.class.php]: Is there any benefit to using sfConfig::get with the second param of default value?  e.g. 

if (sfConfig::get('some_setting', false))

File: apps/qubit/modules/search/templates/descriptionUpdatesSuccess.php [hunk 2]
```diff
@@ -26,7 +26,52 @@
...
+          <td>
+            <?php echo get_component('object', 'clipboardButton', array('slug' => $io->slug, 'wide' => true)) ?>
+          </td>
+        </tr>
+      <?php endforeach; ?>
+      </tbody>
+    </table>
+
+  <?php else: ?>
+  <?php if (isset($pager) && $pager->getNbResults()): ?>
```
Comment (sbreker) [inline apps/qubit/modules/search/templates/descriptionUpdatesSuccess.php]: Indentation formatting needs cleanup.

File: config/schema.yml [hunk 3]
```diff
@@ -44,6 +44,16 @@ propel:
...
 
+  audit_log:
+    id: { type: integer, primaryKey: true, autoIncrement: true, inheritanceKey: false }
+    object_id: { type: integer, required: true, foreignTable: object, foreignReference: id, onDelete: cascade }
+    user_id: { type: integer, required: false, foreignTable: user, foreignReference: id, onDelete: setnull  }
+    user_name: varchar(255)
+    action_type_id: { type: integer, required: false, foreignTable: term, foreignReference: id, onDelete: setnull }
+    created_at: timestamp
+    _indexes:
+      1: [object_id]
```
Comment (sbreker) [inline config/schema.yml]: Are there searches on this table that may need an index other than by object_id?  Do we provide a search like "all changes by user"?  This table could get big.

**Comment by sbreker** [comment]
That's a big change Mike!  Looks great!  I've added a couple questions above.

File: apps/qubit/modules/informationobject/actions/modificationsAction.class.php (hunk 1 above)
Comment (mcantelon) [reply apps/qubit/modules/informationobject/actions/modificationsAction.class.php]: Ah cool... there might be if the setting doesn't exist (maybe it generates a PHP notifice), so I'll put that in to be explicit.

File: config/schema.yml (hunk 3 above)
Comment (mcantelon) [reply config/schema.yml]: This index, and an index for user ID, actually get auto-generated so I've removed this from the schema and updated the migration.

**Comment by sbreker** [comment]
There are a few more of these in your code where the default is not supplied: sfConfig::get('app_audit_log_enabled')

It might make things more clear if the default was supplied in all cases especially if the setting is not yet created.  Not sure if it would result in an error if the setting were not present yet?

**Comment by mcantelon** [comment]
Ah cool... I'll update those calls.

**Comment by sbreker** [comment]
Thanks @mcantelon - Looks great!

**Review by sbreker — APPROVED** [review]
Approved

**Pull request merged**